src/dashboard/src/components/api/views.py

```diff
@@ -824,3 +829,29 @@ def _package_create(request):
         LOGGER.error("{}: {}".format(msg, err))
         return helpers.json_response({"error": True, "message": msg}, 500)
     return helpers.json_response({"id": id_}, 202)
+
+
+@_api_endpoint(expected_methods=["POST"])
+def validate(request):
+    try:
+        body = json.loads(request.body)
+    except Exception:
+        return _error_response("Unexpected request payload")
+    validator_name = body.get("validator")
+    payload = body.get("payload")
+    if not validator_name or not payload:
+        return _error_response("Missing parameters")
+    try:
+        validator = validators.validator(validator_name)
+    except validators.ValidatorNotAvailableError as err:
+        return _error_response(six.text_type(err))
+    try:
+        validator.validate(payload)
+    except validators.ValidationError as err:
+        return _ok_response({"valid": False, "reason": six.text_type(err)})
```

Comment:
This should be HTTP 400 response.

Reply:
I should clarify this comment a bit for posterity — there is a case to be made either way, like all REST API status codes, it's a bit of a judgement call. You can make the argument that validation succeeded and the result was failure, but I think an error status code is more useful for clients if validation failed.